bevm/service_test.go

```diff
@@ -1,11 +1,280 @@
 package bevm
 
 import (
+	"encoding/hex"
+	"encoding/json"
+	"math/big"
+	"strconv"
+	"strings"
 	"testing"
 
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/kyber/v3/suites"
+	"go.dedis.ch/onet/v3"
 	"go.dedis.ch/onet/v3/log"
 )
 
+var tSuite = suites.MustFind("Ed25519")
+
 func TestMain(m *testing.M) {
 	log.MainTest(m)
 }
+
+func setupTest() (*onet.LocalTest, *onet.Roster, *Client) {
+	local := onet.NewTCPTest(tSuite)
+
+	// Generate 1 host, don't connect, process messages, don't register the
+	// tree or entitylist
+	_, ro, _ := local.GenTree(1, false)
+
+	client := &Client{onetClient: local.NewClient(ServiceName)}
+
+	// Do not check leaking goroutines (Ethereum leaves goroutines running...)
+	local.Check = onet.CheckNone
+
+	return local, ro, client
+}
+
+func teardownTest(local *onet.LocalTest) {
```

Comment:
Not really, it was just for symmetry with `setupTest()`...

Reply:
Again, it's how most of `byzcoin/*_test.go` works... If there would be automatic setup- and teardown-methods in go-tests (yes, I know there are packages for this), I think this would be OK.

Reply:
Removed `teardownTest()`